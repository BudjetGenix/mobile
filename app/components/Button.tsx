import React, { FC } from 'react';
import {
  GestureResponderEvent,
  StyleSheet,
  Text,
  TouchableOpacity,
} from 'react-native';

import colors from '@config/colors';

type ButtonProps = {
  title: string;
  onPress?: (event: GestureResponderEvent) => void;
  color?: string;
};

const Button: FC<ButtonProps> = ({
  onPress,
  color = colors.primary,
  title,
}) => {
  return (
    <TouchableOpacity
      onPress={onPress}
      style={[styles.button, { backgroundColor: color }]}
    >
      <Text style={styles.text}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    backgroundColor: colors.primary,
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 15,
    width: '100%',
    marginVertical: 10,
  },
  text: {
    color: colors.white,
    fontSize: 18,
    textTransform: 'uppercase',
    fontWeight: 'bold',
  },
});

export default Button;
