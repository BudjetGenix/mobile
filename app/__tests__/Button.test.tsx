import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../components/Button';

describe('<Button />', () => {
  it('renders properly', () => {
    const btn = renderer.create(<Button />).toJSON();
    expect(btn).toMatchSnapshot();
  });
});
