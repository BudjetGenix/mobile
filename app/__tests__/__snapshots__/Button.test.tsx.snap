// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders properly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderRadius": 25,
      "justifyContent": "center",
      "marginVertical": 10,
      "opacity": 1,
      "padding": 15,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 18,
        "fontWeight": "bold",
        "textTransform": "uppercase",
      }
    }
  />
</View>
`;
